internal/controller/k8s_resources.go

```diff
@@ -20,6 +20,25 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
+// Assuming ServiceAttributes has appropriate fields
+type ServiceAttributes struct {
+	ClusterIPs          []string
+	ExternalIPs         []string
+	LoadBalancerIP      string
+	LoadBalancerIngress []string
+	LoadBalancerClass   string
+	NodePorts           []*Ports
+	ExternalName        string
+	ServiceType         string // make sure you have this field to capture the service type.
+}
```

Comment:
1. Define the non-slice optional fields as pointers.
1. Is `LoadBalancerIngress` really a list? In which case it should be plural.
```suggestion
type ServiceAttributes struct {
	ClusterIPs            []string
	ExternalIPs           []string
	LoadBalancerIP        *string
	LoadBalancerIngresses []string
	LoadBalancerClass     *string
	Ports                 []*Ports
	ExternalName          *string
	Type                  string
}
```

Reply:
Yeah it is a list.